### PR TITLE
Reducers to pure functions

### PR DESCRIPTION
**Pull request by thi4go**
Audit reducers to be pure functions only, and moved logic like assignments and declarations, iterations, to action/lib.

Only two reducers from api module needed a rewrite.

closes https://github.com/decred/politeiagui/issues/562

### CODE REVIEW

File: src/reducers/api.js [hunk 1]
```diff
@@ -125,43 +113,8 @@ export const onResetSyncLikeComment = (state) => {
 };
 
 export const onReceiveSyncLikeComment = (state, action) => {
```
Comment (fernandoabolafio) [inline src/reducers/api.js]: Why did you change this function?

File: src/lib/api.js [hunk 2]
```diff
@@ -228,3 +228,33 @@ export const usernamesById = (csrf, userids) => {
 export const proposalsVoteStatus = () => GET("/v1/proposals/votestatus").then(getResponse);
 export const proposalVoteStatus = (token) => GET(`/v1/proposals/${token}/votestatus`).then(getResponse);
 
+
+export const reduceCommentLikes = (commentsvotes, commentid, token, newAction) => {
```
Comment (fernandoabolafio) [inline src/lib/api.js]: I think it shouldn't be here

File: src/reducers/api.js (hunk 1 above)
Comment (thi4go) [reply src/reducers/api.js]: It had some side effects, like the usage of cloneDeep, some assignments and iterations. Thought It would be better to move everything to the action dispatching it, build the data needed there, and pass down to the reducer.  The actions module where quite clean as well, mostly handling dispatches and callbacks. So I thought that it would be more clean to handle that logic elsewhere, and putted it on lib. 
This is the code that was removed from onReceiveSyncLikeComment

```
const commentsvotes = state.commentsvotes.response &&
    state.commentsvotes.response.commentsvotes;
const backupCV = cloneDeep(commentsvotes);
const comments = state.proposalComments.response &&
    state.proposalComments.response.comments;

  let reducedVotes = null;
  const cvfound = commentsvotes && commentsvotes.find(
    cv => cv.commentid === commentid && cv.token === token
  );

  if (cvfound) {
    reducedVotes = commentsvotes.reduce(
      (acc, cv) => {
        if (cv.commentid === commentid && cv.token === token) {
          const currentAction = parseInt(cv.action, 10);
          acc.oldAction = currentAction;
          cv = {
            ...cv,
            action: newAction === currentAction ? 0 : newAction
          };
        }
        return { ...acc, cvs: acc.cvs.concat([cv]) };
      }, { cvs: [], oldAction: null });
  } else {
    const newCommentVote = { token, commentid, action: newAction };
    reducedVotes = {
      cvs: commentsvotes ? commentsvotes.concat([newCommentVote]) : [newCommentVote],
      oldAction: 0
    };
  }

  const { cvs: newCommentsVotes, oldAction } = reducedVotes;```

**Comment by thi4go** [comment]
Correcting the tests @fernandoabolafio ! Thanks for the review 👍
